### PR TITLE
fix(M2-5702): Rebuilt abTrails component , got rid of deprecated SkiaView component, removed unused code

### PR DESCRIPTION
**Pull request by vmkhitaryanscn**
### 📝 Description

🔗 [Jira Ticket M2-5702](https://mindlogger.atlassian.net/browse/M2-5702)

Changes include:

- Rebuilt abTrails component
- got rid of deprecated SkiaView component
- removed unused code
- Fixed typescript issue in Geolocation item

### ✏️ Notes

SkiaView component was used in AbCanvas component, which was deprecated in the new version of react-native-skia library.

### CODE REVIEW

File: src/shared/ui/SketchCanvas/LineSketcher.ts [hunk 1]
```diff
@@ -85,10 +89,14 @@ class LineSketcher {
...
   public progressLine(
-    path: SkPath,
+    path: SkPath | undefined,
     point: Point,
     straightLine: boolean = false,
   ): void {
+    if (!path) {
+      return;
+    }
+
```
Comment (BamMironov) [inline src/shared/ui/SketchCanvas/LineSketcher.ts:99]: Please elaborate on why the `path` variable can be undefined in the `progressLine` and `addPointToPath`

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/shared/ui/SketchCanvas/LineSketcher.ts (hunk 1 above)
Comment (BamMironov) [reply src/shared/ui/SketchCanvas/LineSketcher.ts:99]: @vmkhitaryanscn I see in this case, I would propose creating a separate method, `.cancel()`. This method should clear the current line and cancel the drawing gesture.

Comment (BamMironov) [reply src/shared/ui/SketchCanvas/LineSketcher.ts:99]: If it is not possible to cancel the gesture, please add a new method to the LineSketcher class called `canProgressLine` that will be responsible for this type of check. Call it before lineSketcher.progressLine. Also, add the check inside the `shouldCreateNewLine` method

File: src/entities/abTrail/ui/AbCanvas.tsx [hunk 2]
```diff
@@ -252,63 +219,59 @@ const AbCanvas: FC<Props> = props => {
...
+    const point: Point = { x, y };
 
     if (isOverAny(point) && !isOverCurrent(point)) {
       onMessage(MessageType.IncorrectStartPoint);
+      resetCurrentPath();
       return;
     }
 
     if (!isOverCurrent(point)) {
+      resetCurrentPath();
```
Comment (anq83) [inline src/entities/abTrail/ui/AbCanvas.tsx:239]: Why the new version contains 3 calls of the "resetCurrentPath" in this function, whereas the previous version doen't have any?

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/entities/abTrail/ui/AbCanvas.tsx [hunk 3]
```diff
@@ -252,63 +219,59 @@ const AbCanvas: FC<Props> = props => {
...
-    if (!isCloseToNext(point)) {
-      resetCloseToNextRerendered();
-    }
-
     onLog({
-      x: (touchInfo.x * width) / 100,
-      y: (touchInfo.y * width) / 100,
-      time: Date.now(),
+      x: (x * width) / 100,
+      y: (x * width) / 100,
```
[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/abTrail/ui/AbCanvas.tsx:273]: Nice catch!

File: src/entities/abTrail/ui/AbCanvas.tsx (hunk 2 above)
Comment (BamMironov) [reply src/entities/abTrail/ui/AbCanvas.tsx:239]: @anq83  here, we leverage line creation to the `SketchCanvas` component. To keep the code as close as possible to the previous solution, it was decided to use `SketchCanvas` as `SkiaView`, which, in turn, automatically resets the drawing line on a touch lift. Hence, to implement such behavior, it is required to reset it manually

File: src/entities/abTrail/ui/AbCanvas.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/abTrail/ui/AbCanvas.tsx:273]: fixed

File: src/entities/abTrail/ui/AbCanvas.tsx [hunk 4]
```diff
@@ -252,63 +219,59 @@ const AbCanvas: FC<Props> = props => {
...
-    if (isCloseToNext(point) && !isCloseToNextRerendered()) {
-      reRender();
-      setCloseToNextRerendered();
-    }
-
-    if (!isCloseToNext(point)) {
-      resetCloseToNextRerendered();
-    }
-
     onLog({
```
Comment (anq83) [inline src/entities/abTrail/ui/AbCanvas.tsx:271]: Please, add local function: addPointToStream and move call of onLog in there for all onLog calls.

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/entities/abTrail/ui/AbCanvas.tsx (hunk 2 above)
Comment (anq83) [reply src/entities/abTrail/ui/AbCanvas.tsx:239]: ok

File: src/entities/abTrail/ui/AbCanvas.tsx (hunk 4 above)
Comment (BamMironov) [reply src/entities/abTrail/ui/AbCanvas.tsx:271]: Done

**Review by anq83 — APPROVED** [review]

**Pull request merged**